libkbfs/block_retrieval_queue.go

```diff
@@ -69,13 +69,16 @@ type blockRetrieval struct {
 	// cancel function for the context
 	cancelFunc context.CancelFunc
 
-	// protects requests and lifetime
+	// protects requests, cacheLifetime, and the prefetch channels
 	reqMtx sync.RWMutex
 	// the individual requests for this block pointer: they must be notified
 	// once the block is returned
 	requests []*blockRetrievalRequest
 	// the cache lifetime for the retrieval
 	cacheLifetime BlockCacheLifetime
+	// channels to notify when this retrieval and its prefetches are done
```

Comment:
I was trying to keep the channel as a semaphore (`chan struct{}` instead of `chan foo`), to avoid the memory overhead of the buffer for each level of the block tree. I commented below regarding how I think I can fix this behavior.